### PR TITLE
fix maven dependency plugin warnings

### LINKED ISSUE #325: Fix dependency-plugin warnings
During mvn verify there are maven-dependency-plugin warnings about Used/Unused declared/undeclared dependencies.

To eliminate them, pom.xml dependencies should be reviewed. 
Also ignore rules in maven-dependency-plugin configuration should be adjusted.

### PR DESCRIPTION
**Pull request by Awambeng**
- add used undeclared dependencies
- remove all unused declared dependencies

### CODE REVIEW

File: datasafe-rest-impl/pom.xml [hunk 1]
```diff
@@ -111,11 +100,6 @@
             </exclusions>
             <scope>runtime</scope>
         </dependency>
-        <dependency>
-            <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-security</artifactId>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml]: It's better add `usedDependency` for spring transitive dependencies, than avoid using spring starters

File: datasafe-simple-adapter/datasafe-simple-adapter-impl/pom.xml [hunk 2]
```diff
@@ -10,6 +10,17 @@
     <modelVersion>4.0.0</modelVersion>
 
     <artifactId>datasafe-simple-adapter-impl</artifactId>
+
+    <properties>
+        <junit.version>4.13.2</junit.version>
```
Comment (max402) [inline datasafe-simple-adapter/datasafe-simple-adapter-impl/pom.xml]: try to replace junit with jupiter

File: datasafe-simple-adapter/datasafe-simple-adapter-spring/pom.xml [hunk 3]
```diff
@@ -14,7 +14,8 @@
     <properties>
         <spring.framework.version>6.1.10</spring.framework.version>
         <spring.framework.boot.version>3.3.1</spring.framework.boot.version>
-        <snakeyaml.version>2.2</snakeyaml.version>
+        <junit.version>4.13.2</junit.version>
```
Comment (max402) [inline datasafe-simple-adapter/datasafe-simple-adapter-spring/pom.xml]: try to replace junit with jupiter

File: last-module-codecoverage-check/pom.xml [hunk 4]
```diff
@@ -112,11 +112,6 @@
             <version>${project.version}</version>
         </dependency>
 
-        <dependency>
-            <groupId>de.adorsys</groupId>
-            <artifactId>datasafe-simple-adapter-api</artifactId>
-            <version>${project.version}</version>
-        </dependency>
```
Comment (max402) [inline last-module-codecoverage-check/pom.xml]: All datasafe modules should be listed in last-module-codecoverage-check's pom.xml. It's needed for jacoco to aggregate reports.

File: datasafe-simple-adapter/datasafe-simple-adapter-spring/pom.xml [hunk 5]
```diff
@@ -135,15 +130,16 @@
...
         <dependency>
             <groupId>jakarta.annotation</groupId>
             <artifactId>jakarta.annotation-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <version>${jetbrains.annotations.version}</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (max402) [inline datasafe-simple-adapter/datasafe-simple-adapter-spring/pom.xml]: This dependency is not needed. all `import org.jetbrains.annotations.` can be replaced by `javax.annotation.`

File: datasafe-simple-adapter/datasafe-simple-adapter-impl/pom.xml [hunk 6]
```diff
@@ -73,6 +118,16 @@
             <artifactId>assertj-core</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <version>${jetbrains.annotations.version}</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (max402) [inline datasafe-simple-adapter/datasafe-simple-adapter-impl/pom.xml]: please remove this dependency

File: datasafe-simple-adapter/datasafe-simple-adapter-impl/pom.xml [hunk 7]
```diff
@@ -40,6 +50,41 @@
...
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-java-sdk-s3</artifactId>
+            <version>${amazon.aws.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-java-sdk-core</artifactId>
+            <version>${amazon.aws.version}</version>
+        </dependency>
```
Comment (max402) [inline datasafe-simple-adapter/datasafe-simple-adapter-impl/pom.xml]: aws dependencies come with datasafe-storage-impl-s3. No need to define them separately.
Better make it as exeption of dependency plugin

File: datasafe-rest-impl/pom.xml [hunk 8]
```diff
@@ -59,29 +65,12 @@
...
-        <dependency>
-            <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-web</artifactId>
-            <version>${spring-boot.version}</version>
-        </dependency>
-        <dependency>
-            <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-actuator</artifactId>
-            <version>${spring-boot.version}</version>
-        </dependency>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml]: Please don't replace starters with all the dependencies they bring

**Review by max402 — COMMENTED** [review]
I don't like that dependency-plugin requires to specify explicitly all transitive dependencies

File: datasafe-rest-impl/pom.xml [hunk 9]
```diff
@@ -19,16 +19,22 @@
...
         <asciidoctor-maven-plugin.version>2.2.4</asciidoctor-maven-plugin.version>
         <exec-maven-plugin.version>1.6.0</exec-maven-plugin.version>
-        <springdoc-openapi-starter-webmvc-ui.version>2.3.0</springdoc-openapi-starter-webmvc-ui.version>
+        <spring.framework.security.version>6.1.2</spring.framework.security.version>
+        <slf4j.version>2.0.7</slf4j.version>
+        <spring.framework.version>6.1.10</spring.framework.version>
+        <spring.framework.restdocs.version>3.0.0</spring.framework.restdocs.version>
+        <jackson.version>2.17.1</jackson.version>
+        <tomcat.version>10.1.11</tomcat.version>
+        <guava.version>32.1.1-jre</guava.version>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml]: guava version already defined in main pom.xml

File: datasafe-simple-adapter/datasafe-simple-adapter-impl/pom.xml [hunk 10]
```diff
@@ -10,6 +10,15 @@
     <modelVersion>4.0.0</modelVersion>
 
     <artifactId>datasafe-simple-adapter-impl</artifactId>
+
+    <properties>
+        <guava.version>32.1.1-jre</guava.version>
```
Comment (max402) [inline datasafe-simple-adapter/datasafe-simple-adapter-impl/pom.xml]: guava version already defined in main pom.xml

File: datasafe-rest-impl/pom.xml [hunk 11]
```diff
@@ -19,16 +19,22 @@
...
+        <gson.version>2.8.9</gson.version>
         <maven.compiler.parameters>true</maven.compiler.parameters>
     </properties>
 
     <dependencies>
-        <dependency>
-            <groupId>org.springdoc</groupId>
-            <artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
-            <version>${springdoc-openapi-starter-webmvc-ui.version}</version>
-        </dependency>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml]: without this dependency swagger doesn't work

File: datasafe-rest-impl/pom.xml [hunk 12]
```diff
@@ -19,16 +19,22 @@
         <spring-restdocs.version>3.0.0</spring-restdocs.version>
         <asciidoctor-maven-plugin.version>2.2.4</asciidoctor-maven-plugin.version>
         <exec-maven-plugin.version>1.6.0</exec-maven-plugin.version>
-        <springdoc-openapi-starter-webmvc-ui.version>2.3.0</springdoc-openapi-starter-webmvc-ui.version>
+        <spring.framework.security.version>6.1.2</spring.framework.security.version>
+        <slf4j.version>2.0.7</slf4j.version>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml]: slf4j version already defined in main pom.xml as slf4j-simple.version

File: datasafe-rest-impl/pom.xml [hunk 13]
```diff
@@ -19,16 +19,22 @@
...
+        <spring.framework.version>6.1.10</spring.framework.version>
+        <spring.framework.restdocs.version>3.0.0</spring.framework.restdocs.version>
+        <jackson.version>2.17.1</jackson.version>
+        <tomcat.version>10.1.11</tomcat.version>
+        <guava.version>32.1.1-jre</guava.version>
+        <jakarta.validation-api.version>3.0.2</jakarta.validation-api.version>
+        <dagger.version>2.50</dagger.version>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml]: dagger version is already defined in main pom.xml

File: datasafe-rest-impl/pom.xml [hunk 14]
```diff
@@ -126,6 +132,71 @@
...
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>${guava.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.validation</groupId>
+            <artifactId>jakarta.validation-api</artifactId>
+            <version>${jakarta.validation-api.version}</version>
+        </dependency>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml:214]: jakarta.validation-api is a transitive dependency of spring-boot-starter-validation

File: datasafe-rest-impl/pom.xml [hunk 15]
```diff
@@ -126,6 +132,71 @@
...
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-java-sdk-s3</artifactId>
+            <version>${amazon.aws.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-java-sdk-core</artifactId>
+            <version>${amazon.aws.version}</version>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml:203]: these aws dependencies are transitive dependencies of datasafe-storage-impl-s3

File: datasafe-rest-impl/pom.xml [hunk 16]
```diff
@@ -126,6 +132,71 @@
...
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-autoconfigure</artifactId>
+            <version>${spring-boot.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.tomcat.embed</groupId>
+            <artifactId>tomcat-embed-core</artifactId>
+            <version>${tomcat.version}</version>
+        </dependency>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml:194]: is part of spring-boot-starter-web

File: datasafe-rest-impl/pom.xml [hunk 17]
```diff
@@ -86,6 +151,12 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-validation</artifactId>
             <version>${spring-boot.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>jakarta.validation</groupId>
+                    <artifactId>jakarta.validation-api</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml]: What is the reason to exclude jakarta.validation-api and then add it separately?

File: datasafe-rest-impl/pom.xml [hunk 18]
```diff
@@ -48,6 +53,16 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>com.amazonaws</groupId>
+                    <artifactId>aws-java-sdk-s3</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>com.amazonaws</groupId>
+                    <artifactId>aws-java-sdk-core</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml]: What is the reason to exclude com.amazonaws dependencies and then add them again?
If there are maven-dependency-plugin warnings, please add ignoredUsedUndeclaredDependencies in plugin configuration.

File: datasafe-rest-impl/pom.xml [hunk 19]
```diff
@@ -64,17 +79,67 @@
...
+                </exclusion>
+                <exclusion>
+                    <groupId>org.apache.tomcat.embed</groupId>
+                    <artifactId>tomcat-embed-core</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>com.fasterxml.jackson.core</groupId>
+                    <artifactId>jackson-databind</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml]: That makes no sense. If there are plugin warnings, then add ignore in plugin configuration.

File: datasafe-rest-impl/pom.xml [hunk 20]
```diff
@@ -64,17 +79,67 @@
...
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-actuator</artifactId>
             <version>${spring-boot.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.springframework.boot</groupId>
+                    <artifactId>spring-boot-actuator-autoconfigure</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml]: please don't exclude from starters, add ignore in plugin configuration

File: datasafe-rest-impl/pom.xml [hunk 21]
```diff
@@ -64,17 +79,67 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>org.springframework.boot</groupId>
+                    <artifactId>spring-boot-autoconfigure</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.springframework.boot</groupId>
+                    <artifactId>spring-boot</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml]: please don't exclude, add ignore in plugin configuration

File: datasafe-rest-impl/pom.xml [hunk 22]
```diff
@@ -115,6 +186,24 @@
...
+                </exclusion>
+                <exclusion>
+                    <groupId>org.springframework.security</groupId>
+                    <artifactId>spring-security-crypto</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.springframework.security</groupId>
+                    <artifactId>spring-security-core</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (max402) [inline datasafe-rest-impl/pom.xml]: please don't exclude, add ignore in plugin configuration

File: datasafe-simple-adapter/datasafe-simple-adapter-impl/pom.xml [hunk 23]
```diff
@@ -28,6 +34,16 @@
...
+            <exclusions>
+                <exclusion>
+                    <groupId>com.amazonaws</groupId>
+                    <artifactId>aws-java-sdk-s3</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>com.amazonaws</groupId>
+                    <artifactId>aws-java-sdk-core</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (max402) [inline datasafe-simple-adapter/datasafe-simple-adapter-impl/pom.xml]: please don't exclude, add ignore in plugin configuration

File: datasafe-simple-adapter/datasafe-simple-adapter-impl/pom.xml [hunk 24]
```diff
@@ -72,6 +133,12 @@
             <groupId>org.assertj</groupId>
             <artifactId>assertj-core</artifactId>
             <scope>test</scope>
+            <exclusions>
+                <exclusion>
+                    <groupId>org.jetbrains</groupId>
+                    <artifactId>annotations</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (max402) [inline datasafe-simple-adapter/datasafe-simple-adapter-impl/pom.xml]: makes no sense. assertj-core doesn't include jetbrains annotation

**Review by max402 — COMMENTED** [review]
Check before commit, that datasafe-rest-impl application is working

**Comment by AssahBismarkabah** [comment]
looks good to me @max402

**Review by AssahBismarkabah — APPROVED** [review]

**Pull request merged**